docx_editor/exceptions.py

```
@@ -68,9 +68,81 @@ class CommentError(DocxEditError):
+        super().__init__(msg)
+
+
+class ParagraphIndexError(DocxEditError):
+    """Raised when a paragraph reference's 1-indexed number is out of range.
+
+    Attributes:
+        index: The paragraph index supplied by the caller (1-indexed).
+        total_paragraphs: Number of paragraphs the document actually has.
+    """
+
+    def __init__(self, index: int, total_paragraphs: int):
```

Ruff (0.15.10)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)